core/iwasm/common/wasm_native.c

```diff
@@ -444,6 +450,9 @@ wasm_native_init()
 #endif
 
 #if WASM_ENABLE_LIB_WASI_THREADS != 0
+    if (!lib_wasi_threads_init())
```

Comment:
Should lib_wasi_threads_destroy() in the handling of fail if lib_wasi_threads_init() is called, e.g. L468, L476? And we should modify L483 to `goto fail`?

Reply:
`lib_wasi_threads_destroy()` is called in `wasm_native_destroy()` so I think that's fine. I agree that L483 should `goto fail` instead of returning, will update that.